common/exception/TypeDBClientException.java

```diff
@@ -98,6 +109,25 @@ private static boolean isUnimplementedMethod(StatusRuntimeException statusRuntim
         return statusRuntimeException.getStatus().getCode() == Status.Code.UNIMPLEMENTED;
     }
 
+    private static TypeDBClientException tryGetRstStreamCause(StatusRuntimeException sre) {
+        Throwable cause = null;
+        if (sre.getStatus().getCode() == Status.Code.UNAVAILABLE) {
+            if (sre.getCause() instanceof javax.net.ssl.SSLHandshakeException) {
+                if (sre.getCause().getMessage().contains("PKIX path building failed")) {
+                    return new TypeDBClientException(sre.getCause(), CLUSTER_SSL_CERTIFICATE_NOT_VALIDATED);
+                } else if (sre.getCause().getMessage().contains("No subject alternative DNS name")) {
+                    return new TypeDBClientException(sre.getCause(), CLUSTER_SSL_CERTIFICATE_INVALID_FOR_HOSTNAME);
+                } else return new TypeDBClientException(sre.getCause(), CLUSTER_SSL_HANDSHAKE_FAILED, sre.getMessage());
```

Comment:
I introduce an exception for this branch. Should I let it fall through to the end instead?

Reply:
This seems ok to me!